stream/src/main/java/org/apache/kafka/stream/KStream.java

```diff
@@ -150,7 +134,7 @@
     /**
      * Processes all elements in this stream by applying a processor.
      *
-     * @param processor the instance of Processor
+     * @param processorClass the class of Processor
      */
-    <K1, V1> KStream<K1, V1> process(KafkaProcessor<K, V, K1, V1> processor);
+    <K1, V1> KStream<K1, V1> process(Class<? extends KafkaProcessor<K, V, K1, V1>> processorClass, PConfig config);
```

Comment:
Also note that this class object passing makes it impossible to mimic Spark Streaming API which uses a regular instance passing.

Umm, doesn't this imply that having additional layer under KStream is not helping developers to define a new API???